flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/ReactiveModeUtils.java

```diff
@@ -47,5 +51,20 @@ public static void configureJobGraphForReactiveMode(JobGraph jobGraph) {
         }
     }
 
+    public static void configureClusterForReactiveMode(Configuration configuration) {
+        LOG.info("Modifying Cluster configuration for reactive mode");
+
+        if (!configuration.contains(JobManagerOptions.RESOURCE_STABILIZATION_TIMEOUT)) {
+            // Configure adaptive scheduler to schedule job even if desired resources are not
+            // available (but sufficient resources)
+            configuration.set(
+                    JobManagerOptions.RESOURCE_STABILIZATION_TIMEOUT, Duration.ofMillis(0));
+        }
+        if (!configuration.contains(JobManagerOptions.RESOURCE_WAIT_TIMEOUT)) {
+            // configure adaptive scheduler to wait forever for TaskManagers to register
+            configuration.set(JobManagerOptions.RESOURCE_WAIT_TIMEOUT, Duration.ofMillis(-1));
+        }
+    }
```

Comment:
I am not saying that we have to change it but I wanted to make an observation about this kind of pattern (changing the `Configuration` so that the right configuration values are being picked up later):

The problem I see with this pattern is that it decouples the instantiation of the required components and their configuration. Consequently, we as the developers have to remember that at every entrypoint which might use the `AdaptiveScheduler` we have to call `ReactiveModeUtils.configureclusterForReactiveMode` (here we have the `ClusterEntrypoint` and the `MiniCluster`). If we forget it, then the reactive mode might not really work.

If these context sensitive configuration option overwrites would be moved closer to the instantiation of the components,  then it is easier to reuse the "reactive mode" in another setting. E.g. when seeing in the `DefaultJobManagerRunnerFactory` that we are running a job in reactive mode, then one could read `RESOURCE_STABILIZATION_TIMEOUT` and `RESOURCE_WAIT_TIMEOUT` with different default values (`0` and `-1`) which are then passed to the factory of the `AdaptiveScheduler`.

Reply:
I think the key observation is that if we keep related things closer together, then it is easier to reuse them and it is easier to maintain them because they are not spread all over the place.